config.js

```diff
@@ -1,72 +1,84 @@
+const { getXApiKey } = require('./utils/x-api-key-settings.js');
+
 const CONTRACT_NAME = process.env.CONTRACT_NAME;
 
 function getConfig(env) {
+    let config;
     switch (env) {
-
     case 'production':
     case 'mainnet':
-        return {
+        config = {
             networkId: 'mainnet',
-            nodeUrl: 'https://rpc.mainnet.near.org',
+            nodeUrl: process.env.NEAR_CLI_MAINNET_RPC_SERVER_URL || 'https://rpc.mainnet.near.org',
             contractName: CONTRACT_NAME,
             walletUrl: 'https://wallet.near.org',
             helperUrl: 'https://helper.mainnet.near.org',
             helperAccount: 'near',
             explorerUrl: 'https://explorer.mainnet.near.org',
         };
+        break;
     case 'development':
     case 'testnet':
-        return {
+        config = {
             networkId: 'testnet',
-            nodeUrl: 'https://rpc.testnet.near.org',
+            nodeUrl: process.env.NEAR_CLI_TESTNET_RPC_SERVER_URL || 'https://rpc.testnet.near.org',
             contractName: CONTRACT_NAME,
             walletUrl: 'https://wallet.testnet.near.org',
             helperUrl: 'https://helper.testnet.near.org',
             helperAccount: 'testnet',
             explorerUrl: 'https://explorer.testnet.near.org',
         };
+        break;
     case 'betanet':
-        return {
+        config = {
             networkId: 'betanet',
-            nodeUrl: 'https://rpc.betanet.near.org',
+            nodeUrl: process.env.NEAR_CLI_BETANET_RPC_SERVER_URL || 'https://rpc.betanet.near.org',
             contractName: CONTRACT_NAME,
             walletUrl: 'https://wallet.betanet.near.org',
             helperUrl: 'https://helper.betanet.near.org',
             helperAccount: 'betanet',
             explorerUrl: 'https://explorer.betanet.near.org',
         };
+        break;
     case 'guildnet':
-        return {
+        config = {
             networkId: 'guildnet',
-            nodeUrl: 'https://rpc.openshards.io',
+            nodeUrl: process.env.NEAR_CLI_GUILDNET_RPC_SERVER_URL || 'https://rpc.openshards.io',
             contractName: CONTRACT_NAME,
             walletUrl: 'https://wallet.openshards.io',
             helperUrl: 'https://helper.openshards.io',
             helperAccount: 'guildnet',
         };
+        break;
     case 'local':
     case 'localnet':
-        return {
+        config = {
             networkId: process.env.NEAR_CLI_LOCALNET_NETWORK_ID || 'local',
-            nodeUrl: process.env.NEAR_NODE_URL || 'http://localhost:3030',
+            nodeUrl: process.env.NEAR_CLI_GUILDNET_RPC_SERVER_URL || process.env.NEAR_NODE_URL || 'http://localhost:3030',
```

Comment:
Did you actually mean `NEAR_CLI_LOCALNET_RPC_SERVER_URL`?

Reply:
The same :) Thank you!